cmds/db-manager/main.go

```diff
@@ -0,0 +1,218 @@
+		sslDir:  flag.String("cockroach_ssl_dir", "", "directory to ssl certificates. Must contain files: ca.crt, client.<user>.crt, client.<user>.key"),
+	}
+)
+
+func main() {
+	flag.Parse()
+	if *path == "" {
+		log.Panic("Must specify migration_files path")
+	}
+	if (*dbVersion == "" && *step == 0) || (*dbVersion != "" && *step != 0) {
+		log.Panic("Must specify a db_version or migration_step to goto")
+	}
+	uriParams := map[string]string{
+		"host":             *cockroachParams.host,
+		"port":             strconv.Itoa(*cockroachParams.port),
+		"user":             *cockroachParams.user,
+		"ssl_mode":         *cockroachParams.sslMode,
+		"ssl_dir":          *cockroachParams.sslDir,
+		"application_name": "SchemaManager",
+	}
+	postgresURI, err := dssCockroach.BuildURI(uriParams)
+	if err != nil {
+		log.Panic("Failed to build URI", zap.Error(err))
+	}
+	crdbURI := strings.Replace(postgresURI, "postgresql", "cockroachdb", 1)
+	*path = fmt.Sprintf("file://%v", *path)
+	migrater, migraterErr := migrate.New(*path, crdbURI)
+	myMigrater := &MyMigrate{migrater}
+	defer func() {
+		if migraterErr == nil {
+			if _, err := myMigrater.Close(); err != nil {
+				log.Println(err)
+			}
+		}
+	}()
+	if migraterErr == nil {
+		myMigrater.PrefetchMigrations = 10
+		myMigrater.LockTimeout = time.Duration(15) * time.Second
+
+		// handle Ctrl+c
+		signals := make(chan os.Signal, 1)
+		signal.Notify(signals, syscall.SIGINT)
+		go func() {
+			for range signals {
+				log.Println("Stopping after this running migration ...")
+				myMigrater.GracefulStop <- true
+				return
+			}
+		}()
+	}
+	totalMove := 0
+	direction := myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	for direction != 0 {
+		executeErr := myMigrater.Steps(direction)
+		if executeErr != nil {
+			log.Fatal(executeErr)
+		}
+		totalMove += direction
+		log.Printf("Migrated %s by %d step", directionText(direction), int(math.Abs(float64(direction))))
+		direction = myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	}
+	if totalMove == 0 {
+		log.Println("No Changes")
+	} else {
+		log.Printf("Moved %d steps %s in total", int(math.Abs(float64(totalMove))), directionText(totalMove))
+	}
+
+	currentDBVersion, err := getCurrentDBVersion(postgresURI)
+	if err != nil {
+		log.Fatal("Failed to get Current DB version for confirmation")
+	}
+	migrationStep, dirty, err := myMigrater.Version()
+	if err != nil {
+		log.Fatal("Failed to get Migration Step for confirmation")
+	}
+	log.Printf("DB Version: %s, Migration Step # %d, Dirty: %v", currentDBVersion, migrationStep, dirty)
+}
+
+func directionText(direction int) string {
+	if direction > 0 {
+		return "Up"
+	} else if direction < 0 {
+		return "Down"
+	}
+	return "No Change"
+}
+
+func getCurrentDBVersion(crdbURI string) (string, error) {
+	crdb, err := dssCockroach.Dial(crdbURI)
+	if err != nil {
+		log.Panic("Failed to dial CockroachDB")
+	}
+	// check if schema_versions table exists
```

Comment:
Please add `defer crdb.Close()` here, and remove the manual close by the end of the function.